pkg/controller/volume/persistentvolume/pv_controller.go

```diff
@@ -918,6 +926,35 @@ func (ctrl *PersistentVolumeController) updateVolumePhaseWithEvent(volume *v1.Pe
 	return newVol, nil
 }
 
+// assignDefaultStorageClass looks up all storage classes and attempts to assign a storage class to the given claim.
+// The storage class assigned is always the default one and if there is none or there are more defaults the claim
+// will not get updated.
+func (ctrl *PersistentVolumeController) assignDefaultStorageClass(claim *v1.PersistentVolumeClaim) (*v1.PersistentVolumeClaim, error) {
```

Comment:
This will get called when a PVC is updated or when the periodic sync fires.  How often is that?  Is it good enough?  Or should we also watch StorageClasses and trigger something whenever the effective default changes?  That's a lot more work, but if this will only trigger every O(days) it seems kind of bad..

Reply:
Resync is every 15 seconds, and unfortunately there is behavior that is depending on frequent resyncs to happen rather than setting up the appropriate watches. I would love to get rid of the resync some day... https://github.com/kubernetes/kubernetes/issues/85661

Reply:
Sad!  :)

Can we add a comment somewhere that spells out "if we ever change the resync to a larger period, we might need to change how we set the default class on existing unbound claims".